### PR TITLE
Explore page

### PR DESCRIPTION
**Pull request by nina-dk**
I've still got some work to do but most of it is done I think. Let me know your thoughts.

### CODE REVIEW

**Review by radu-constantin — APPROVED** [review]
I simply love those animations. They look incredible! 
It's starting to look very polished.
For the rest of the page I have the following suggestions:
1. Could we make the Neighborhood cards have the same shape as in the HomePage (wider);
2. When no neighborhood is shown after searching can we display a message such as (No neighborhoods were found with that name? or something similar);
As far as the code goes, I didn't find any problems. I will approve the PR and maybe you can make the changes later.

File: apps/frontend/src/components/NeighborhoodSearch/NeighborhoodSearch.tsx [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    } catch (error) {
+      if (error instanceof AxiosError) {
+        const errorResponse: ErrorObj = error.response?.data;
+        setErrorMsg(errorResponse.error);
+      } else if (error instanceof Error) {
+        setErrorMsg(error.message);
+      }
+
+      window.scrollTo(0, 0);
+      console.error(error);
```
Comment (radu-constantin) [inline apps/frontend/src/components/NeighborhoodSearch/NeighborhoodSearch.tsx:59]: I don't think this is needed anymore

File: apps/frontend/src/components/NeighborhoodSearch/NeighborhoodSearch.tsx [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+  neighborhoods,
+  cursor,
+  isNextPage,
+}: {
+  neighborhoods: Neighborhood[];
+  cursor?: number;
+  isNextPage: boolean;
+}) {
+  const [neighborhoodList, setNeighborhoodList] = useState(neighborhoods);
+  const [hasNextPage, setHasNextPage] = useState(isNextPage);
```
Comment (radu-constantin) [inline apps/frontend/src/components/NeighborhoodSearch/NeighborhoodSearch.tsx:27]: Does changing this need to trigger a re-render?

File: apps/frontend/src/components/NeighborhoodSearch/NeighborhoodSearch.tsx [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+  const [currentCursor, setCurrentCursor] = useState(cursor);
```
Comment (radu-constantin) [inline apps/frontend/src/components/NeighborhoodSearch/NeighborhoodSearch.tsx:28]: Same here

File: apps/frontend/src/components/NeighborhoodSearch/NeighborhoodSearch.tsx [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    let filteredNeighborhoods: Neighborhood[] | null = [];
+
+    const timeout = setTimeout(async () => {
+      if (searchTerm.length > 0) {
+        setErrorMsg('');
```
Comment (radu-constantin) [inline apps/frontend/src/components/NeighborhoodSearch/NeighborhoodSearch.tsx:70]: Why is this needed?

**Review by radu-constantin — CHANGES_REQUESTED** [review]
Incredible work Antonina! I have a few questions regarding the code, but other than that, this is a very polished feature!

File: apps/frontend/src/hooks/usePagination.ts
[the remark this answers is not in the document]
Comment (nina-dk) [reply apps/frontend/src/hooks/usePagination.ts:1]: Nop! I was initially going to use pagination and created this. Will get rid of it.

File: apps/frontend/src/components/NeighborhoodSearch/NeighborhoodSearch.tsx (hunk 1 above)
Comment (nina-dk) [reply apps/frontend/src/components/NeighborhoodSearch/NeighborhoodSearch.tsx:59]: Are you talking about the `console.error`? I guess not

File: apps/frontend/src/components/NeighborhoodSearch/NeighborhoodSearch.tsx (hunk 4 above)
Comment (nina-dk) [reply apps/frontend/src/components/NeighborhoodSearch/NeighborhoodSearch.tsx:70]: It's not strictly needed, but if an error is raised from `fetchData`, for example, and then you do a search, the error remains at the top even though there is no issue with the filtering functionality.

File: apps/frontend/src/components/NeighborhoodSearch/NeighborhoodSearch.tsx (hunk 2 above)
Comment (nina-dk) [reply apps/frontend/src/components/NeighborhoodSearch/NeighborhoodSearch.tsx:27]: Its value is passed to `InfiniteScroll` so I'd think yes.

File: apps/frontend/src/components/NeighborhoodSearch/NeighborhoodSearch.tsx (hunk 3 above)
Comment (nina-dk) [reply apps/frontend/src/components/NeighborhoodSearch/NeighborhoodSearch.tsx:28]: hmm not sure tbh. What's your proposed alternative?

File: apps/frontend/src/components/NeighborhoodSearch/NeighborhoodSearch.tsx (hunk 4 above)
Comment (radu-constantin) [reply apps/frontend/src/components/NeighborhoodSearch/NeighborhoodSearch.tsx:70]: Got it!

File: apps/frontend/src/components/NeighborhoodSearch/NeighborhoodSearch.tsx (hunk 1 above)
Comment (radu-constantin) [reply apps/frontend/src/components/NeighborhoodSearch/NeighborhoodSearch.tsx:59]: Yep, the console error

File: apps/frontend/src/components/NeighborhoodSearch/NeighborhoodSearch.tsx (hunk 3 above)
Comment (radu-constantin) [reply apps/frontend/src/components/NeighborhoodSearch/NeighborhoodSearch.tsx:28]: I'd propose we use https://react.dev/reference/react/useRef in the future when we need to store data in components that do not need to trigger a re-render.
In this case, I think 'hasNextPage' and 'currentCursor' do not need to trigger a re-render.

**Review by radu-constantin — APPROVED** [review]
You've done an amazing job again, Antonina! Congratulations!

File: apps/frontend/src/components/NeighborhoodSearch/NeighborhoodSearch.tsx (hunk 3 above)
Comment (nina-dk) [reply apps/frontend/src/components/NeighborhoodSearch/NeighborhoodSearch.tsx]: @radu-constantin I've made the updates. Check it out when you have the time!

**Comment by radu-constantin** [comment]
Great job Antonina! You can merge whenever you're ready!
